### PR TITLE
Move public Waveplot variables to derived types

### PR DESCRIPTION
**Pull request by vanderhe**
Since DFTB+ moves away from public variables, I thought this might be a useful addition to keep the code consistent. The introduced derived types are based on the individual input blocks. If a different classification is desired, just let me know.

### CODE REVIEW

**Review by awvwgk — COMMENTED** [review]
Minor comment regarding naming, ``TXml`` and ``xml`` could probably named ``TInput``and ``input`` to make their intent clearer.

File: prog/waveplot/initwaveplot.F90 [hunk 1]
```diff
@@ -252,133 +320,158 @@ subroutine initProgramVariables()
...
+            & * kPointsWeights(4, iKpoint)
       end do
     end if
 
   end subroutine readDetailed
 
 
   !> Read in the geometry stored as xml in internal or gen format.
-  subroutine readGeometry(geonode, geo)
+  subroutine readGeometry(this, geonode)
```
[the remark this answers is not in the document]
Comment (vanderhe) [reply prog/waveplot/initwaveplot.F90:428]: You are right, I will correct that.

[the remark this answers is not in the document]
Comment (vanderhe) [reply prog/waveplot/initwaveplot.F90:428]: I will also rename `TXml` and `xml` according to your suggestion.

**Review by awvwgk — APPROVED** [review]
Thanks, looks good to me now.

**Pull request merged**